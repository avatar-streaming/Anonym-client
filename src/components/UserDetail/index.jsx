import React from "react";
import useUserName from "../../hooks/useUserName";
import useUpdateUserName from "../../hooks/useUpdateUserName";
import { useSelector } from "react-redux";

function UserDetail() {
  const [userName, updateUserName] = useUserName();
  const isUpdateUserName = useUpdateUserName(userName);
  const { thumnail } = useSelector((state) => state.auth.userInfo);

  return (
    <div className="content-wrapper">
      <div>
        <h3>Profile</h3>
      </div>
      <div>
        <h4>Profile Picture</h4>
        <img className="user-thumnail" src={thumnail} alt="user thumnail" />
        <form>
          <input type="file" accept="image/*" />
          <button>save</button>
        </form>
      </div>
      <div>
        <div>
          <h4>Profile Setting</h4>
          <div>
            <h5>Username</h5>
            <form>
              <input
                type="text"
                className="input-text"
                value={userName}
                onChange={(e) => {
                  updateUserName(e.target.value.trim());
                }}
              />
              <button onClick={(e) => {
                e.preventDefault();
                isUpdateUserName(true);
              }}>
                save
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
}

export default UserDetail;
