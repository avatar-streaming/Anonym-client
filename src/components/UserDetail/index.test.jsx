import React from "react";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import UserDetail from ".";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";

describe("<UserDetail />", () => {
  const mockStore = configureMockStore([thunk]);
  let store = null;
  let component = null;

  beforeEach(() => {
    store = mockStore({
      user: {
        userInfo: {
          userName: "mock name 1",
          thumnail: "mock thumnail 1",
        },
      },
    });
    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <UserDetail />
      </Provider>
    );
  });

  it("should render with given state from redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should update user thumnail", () => {
    const userThumnailInput = component.root.findByProps({ type: "file" });
    const userThumnailButton = component.root.findByProps({ "data-testid": "thumnail" });

    renderer.act(() => {
      userThumnailInput.props.onChange({
        target: {
          value: "mock thumnail 2",
        },
      });
    });
    expect(component.toJSON()).toMatchSnapshot();

    renderer.act(() => {
      userThumnailButton.props.onClick({
        preventDefault: jest.fn(),
        isUpdateUserThumnail: store.dispatch(),
      });
    });
    expect(store.dispatch).toHaveBeenCalledTimes(1);
  });

  it("should update user name", () => {
    const userNameInput = component.root.findByProps({ type: "text" });
    const userNameButton = component.root.findByProps({ "data-testid": "name" });

    renderer.act(() => {
      userNameInput.props.onChange({
        target: {
          value: "mock name 2 ",
        },
      });
    });
    expect(component.toJSON()).toMatchSnapshot();

    renderer.act(() => {
      userNameButton.props.onClick({
        preventDefault: jest.fn(),
        isUpdateUserName: store.dispatch(),
      });
    });
    expect(store.dispatch).toHaveBeenCalledTimes(1);
  });
});
