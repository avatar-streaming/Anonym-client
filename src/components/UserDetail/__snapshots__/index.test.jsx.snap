// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDetail /> should dispatch an action on button click 1`] = `
<div
  className="content-wrapper"
>
  <div>
    <h3
      className="profile-title"
    >
      Profile
    </h3>
  </div>
  <div>
    <h4>
      Profile Picture
    </h4>
    <img
      alt="user thumnail"
      className="user-thumnail"
      src="mock thumnail"
    />
    <form>
      <input
        accept="image/*"
        type="file"
      />
      <button>
        save
      </button>
    </form>
  </div>
  <div>
    <div>
      <h4>
        Profile Setting
      </h4>
      <div>
        <h5
          className="sub-title"
        >
          Username
        </h5>
        <form>
          <input
            className="input-text"
            onChange={[Function]}
            type="text"
            value="mock name 2"
          />
          <button
            className="user-name-button"
            onClick={[Function]}
          >
            save
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`<UserDetail /> should render with given state from redux store 1`] = `
<div
  className="content-wrapper"
>
  <div>
    <h3
      className="profile-title"
    >
      Profile
    </h3>
  </div>
  <div>
    <h4>
      Profile Picture
    </h4>
    <img
      alt="user thumnail"
      className="user-thumnail"
      src="mock thumnail"
    />
    <form>
      <input
        accept="image/*"
        type="file"
      />
      <button>
        save
      </button>
    </form>
  </div>
  <div>
    <div>
      <h4>
        Profile Setting
      </h4>
      <div>
        <h5
          className="sub-title"
        >
          Username
        </h5>
        <form>
          <input
            className="input-text"
            onChange={[Function]}
            type="text"
            value="mock name 1"
          />
          <button
            className="user-name-button"
            onClick={[Function]}
          >
            save
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
